community/common/src/main/java/org/neo4j/function/Predicates.java

```diff
@@ -144,7 +144,15 @@ public boolean test( T item )
     public static void await( Supplier<Boolean> condition, long timeout, TimeUnit unit )
             throws TimeoutException, InterruptedException
     {
-        long sleep = Math.max( unit.toMillis( timeout ) / 100, 1 );
+        long checkIntervalMillis = Math.max( unit.toMillis( timeout ) / 100, 10 );
+        long timeoutMillis = unit.toMillis( timeout );
+        await( condition, timeoutMillis, checkIntervalMillis, TimeUnit.MILLISECONDS );
+    }
+
+    public static void await( Supplier<Boolean> condition, long timeout, long checkInterval, TimeUnit unit )
```

Comment:
What do you think if we will call checkInterval - sleepInterval instead and will specify in javadoc that unit, timeout and sleepInterval expected to represent values more then 1 millisecond with corresponding unit?.